### PR TITLE
Clear test data for applications the current provider ratifies.

### PR DESCRIPTION
**Pull request by steventux**
## Context

The API endpoint `test-data/clear` clears applications for courses run by the current provider.
It also needs to clear applications data for courses accredited/ratified by the current provider. 

## Changes proposed in this pull request

Adds an additional step in the test data controller `clear!` action which destroys any application data (Candidate, and dependents) where the current provider ratifies the course.
 

## Link to Trello card

https://trello.com/c/y1mz1pb3/3494-api-endpoint-for-test-data-clear-does-not-clear-data-for-courses-the-provider-ratifies

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/vendor_api/test_data_controller.rb [hunk 1]
```diff
@@ -56,5 +57,12 @@ def courses_per_application_param
     def for_ratified_courses_param
       params[:for_ratified_courses] == 'true'
     end
+
+    def clear_accredited_application_data!
+      Candidate
```
Comment (cwrw) [inline app/controllers/vendor_api/test_data_controller.rb:62]: I'll attempt one query for both cases, will push the change

File: app/models/candidate.rb [hunk 2]
```diff
@@ -22,6 +22,12 @@ def self.for_email(email)
     find_or_initialize_by(email_address: email.downcase) if email
   end
 
+  def self.clear_application_data!(provider)
+    scope = joins(application_forms: { application_choices: { course_option: :course } })
+    scope.where("courses.accredited_provider": provider).or(scope.where("courses.provider": provider))
+      .delete_all
+  end
+
```
Comment (benswannack) [inline app/models/candidate.rb:30]: Could we move this out to a small service?

Comment (cwrw) [reply app/models/candidate.rb]: done

**Review by benswannack — APPROVED** [review]
LGTM!

**Pull request merged**